Mathlib/Algebra/Order/Ring/Unbundled/Rat.lean

```diff
@@ -125,21 +120,11 @@ instance : AddLeftMono ℚ where
 theorem div_lt_div_iff_mul_lt_mul {a b c d : ℤ} (b_pos : 0 < b) (d_pos : 0 < d) :
     (a : ℚ) / b < c / d ↔ a * d < c * b := by
   simp only [lt_iff_le_not_ge]
-  apply and_congr
-  · simp [div_def', Rat.divInt_le_divInt b_pos d_pos]
-  · simp [div_def', Rat.divInt_le_divInt d_pos b_pos]
+  simp [*, div_def', Rat.divInt_le_divInt]
 
 theorem lt_one_iff_num_lt_denom {q : ℚ} : q < 1 ↔ q.num < q.den := by simp [Rat.lt_iff]
 
 theorem abs_def (q : ℚ) : |q| = q.num.natAbs /. q.den := by
-  rcases le_total q 0 with hq | hq
-  · rw [abs_of_nonpos hq]
-    rw [← num_divInt_den q, ← zero_divInt, Rat.divInt_le_divInt (mod_cast q.pos) Int.zero_lt_one,
-      mul_one, zero_mul] at hq
-    rw [Int.ofNat_natAbs_of_nonpos hq, ← neg_def]
-  · rw [abs_of_nonneg hq]
-    rw [← num_divInt_den q, ← zero_divInt, Rat.divInt_le_divInt Int.zero_lt_one (mod_cast q.pos),
-      mul_one, zero_mul] at hq
-    rw [Int.natAbs_of_nonneg hq, num_divInt_den]
+  grind [abs_of_nonpos, neg_def, Rat.num_nonneg, abs_of_nonneg, num_divInt_den]
```

Comment:
Replacing proofs by `grind` applications seems generally accepted by the maintainers, and it does feel less of a slog here.